gnd/src/main/java/com/google/android/gnd/persistence/local/room/entity/LayerEntity.java

```diff
@@ -92,22 +93,31 @@ static Layer toLayer(LayerEntityAndRelations layerEntityAndRelations) {
       layerBuilder.setForm(FormEntity.toForm(formEntityAndRelations));
     }
 
-    layerBuilder.setContributorsCanAdd(toList(layerEntity.getContributorsCanAdd()));
+    layerBuilder.setContributorsCanAdd(toFeatureTypes(layerEntity.getContributorsCanAdd()));
 
     return layerBuilder.build();
   }
 
-  private static ImmutableList<String> toList(JSONArray jsonArray) {
-    ImmutableList.Builder builder = ImmutableList.builder();
+  private static ImmutableList<FeatureType> toFeatureTypes(JSONArray jsonArray) {
+    ImmutableList.Builder<FeatureType> builder = ImmutableList.builder();
     for (int i = 0; i < jsonArray.length(); i++) {
       String value = jsonArray.optString(i, null);
       if (value != null) {
-        builder.add(value);
+        builder.add(toFeatureType(value));
       }
     }
     return builder.build();
   }
 
+  private static FeatureType toFeatureType(String stringValue) {
+    switch (stringValue) {
+      case "points":
+        return FeatureType.POINT;
```

Comment:
It would be simpler to convert string -> enum using `toEnum()` if both stringValue and Enum were either "point" or "points".

Reply:
It would be less code, but without making the mapping explicit, a rename of the enum would break the code. Ideally the enum values should be named for what's more readable in code, and the values in the db for whatever makes most sense there. Coupling them could force a compromise on either end.